src/kakarot/instructions/memory_operations.cairo

```diff
@@ -219,16 +220,32 @@ namespace MemoryOperations {
     }(evm: model.EVM*) -> model.EVM* {
         alloc_locals;
 
+        let (popped) = Stack.pop_n(2);
```

Comment:
can we pop after the checks and early returns?

Reply:
Values should be popped off the stack before any other operation!